### DISCUSSION TITLE
How to manually login a user?

### DISCUSSION THREAD

**Discussion by warvariuc** [fastapi-users/fastapi-users#558] [Q&A] [answered]
I have an endpoint in which I would like to log in a user by an access code.

How do I do that?

**Comment by frankie567** [comment, marked as answer]
It depends on what you mean by "log in" a user. If you are looking for generating a JWT or a Cookie, you can for example call the `get_login_response` method of your authentication object. Something like:

```py
@app.post("/auth/code/{code}")
async def code_login(code: str, response: Response):
  user = await get_user_from_code(code)
  return jwt_authentication.get_login_response(user, response)
```
